internal/list.go

```diff
@@ -11,3 +33,320 @@ func newListProxy(client *HazelcastClient, serviceName *string, name *string) (*
 	}
 	return &ListProxy{parSpecProxy}, nil
 }
+
+func (list *ListProxy) Add(element interface{}) (changed bool, err error) {
+	if element == nil {
+		return false, core.NewHazelcastNilPointerError(NIL_VALUE_IS_NOT_ALLOWED, nil)
+	}
+	valueData, err := list.ToData(element)
+	if err != nil {
+		return false, err
+	}
+	request := ListAddEncodeRequest(list.name, valueData)
+	responseMessage, err := list.Invoke(request)
+	if err != nil {
+		return false, err
+	}
+	changed = ListAddDecodeResponse(responseMessage).Response
+	return changed, nil
+}
+
+func (list *ListProxy) AddAt(index int32, element interface{}) (err error) {
+	if element == nil {
+		return core.NewHazelcastNilPointerError(NIL_VALUE_IS_NOT_ALLOWED, nil)
+	}
+	valueData, err := list.ToData(element)
+	if err != nil {
+		return err
+	}
+	request := ListAddWithIndexEncodeRequest(list.name, index, valueData)
+	_, err = list.Invoke(request)
+	return err
+}
+
+func (list *ListProxy) AddAll(elements []interface{}) (changed bool, err error) {
+	if len(elements) == 0 {
+		return false, core.NewHazelcastNilPointerError(NIL_SLICE_IS_NOT_ALLOWED, nil)
```

Comment:
`len(elements)` being 0 does not necessarily mean `elements` is `nil`. `elements` could be an empty slice, right? In that case I should be able to call `AddAll` with an empty `elements`.

Reply:
Elements are checked separately if any of them is nil. Like if a slice has 4 elements and one nil element then it will also return an error. Is that what you were saying ?

Reply:
```go
emptySlice := make([]int, 0, 10)
changed, err = :list.AddAll(emptySlice);
```
should not set `err`. Above is a valid operation that should return `false`. In current implementation it sets `err`.